File: game-db/src/main/java/com/jacey/game/db/entity/PlayStateEntity.java
package com.jacey.game.db.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @Description: 玩家状态
 * @Author: JaceyRuan
 * @Email: jacey.ruan@outlook.com
 */
@Data
@Document(collection = PlayStateEntity.COLLECTION_NAME)
public class PlayStateEntity {

    public static final String COLLECTION_NAME = "play_state";
    @Id
    private int id;

    private int userId;

    /** 在线状态（在线or离线）  */
    private int userOnlineState;

    /** 行为状态（none or 匹配中 or 对战中 等） */
    private int userActionState;

    /** 对战类型（1v1 or ..） */
    private int battleType;

    /** 对战id */
    private String battleId;
}

File: game-db/src/main/kotlin/com/jacey/game/db/entity/PlayStateEntity.kt
package com.jacey.game.db.entity

import com.jacey.game.common.framework.mongo.DbDocument

/**
 * 玩家状态（Mongo 集合 play_state）
 */
data class PlayStateEntity(
    override val _id: Int,
    var userId: Int = 0,
    /** 在线状态（在线or离线） */
    var userOnlineState: Int = 0,
    /** 行为状态（none or 匹配中 or 对战中 等） */
    var userActionState: Int = 0,
    /** 对战类型（1v1 or ..） */
    var battleType: Int = 0,
    /** 对战id */
    var battleId: String? = null,
) : DbDocument<Int> {
    val id: Int get() = _id

    companion object {
        const val COLLECTION_NAME = "play_state"
    }
}